### PR TITLE
Readme & package updates

### PR DESCRIPTION
**Pull request by zone117x**
* Updated blockstack.js and bip js dependencies -- required refactoring several sync methods to async. 
* Removed several unused dependencies. 
* Fixed lint process and all lint errors. 
* Improved installation process and updated readme.
* Added a usage example to readme.

Tested with blockstack-core integration tests:
* Fixed the docker image and pointed to it to this feature branch in this commit: https://github.com/blockstack/blockstack-core/commit/f3cd36f3416cbb9b01832da89d0eba966bfd5fed
* Ran `$ docker build -f integration_tests/deployment/docker/Dockerfile.tests -t quay.io/blockstack/integrationtests:matt .`
* Then 
   ```
   $ IMAGE=$(docker run -dt -v /tmp:/tmp quay.io/blockstack/integrationtests:matt blockstack-test-scenario blockstack_integration_tests.scenarios.name_pre_reg)
   $ docker logs -f $IMAGE
   ...
   SUCCESS blockstack_integration_tests.scenarios.name_pre_reg
```

### CODE REVIEW

File: src/auth.ts [hunk 1]
```diff
@@ -90,36 +85,36 @@ export interface NamedIdentityType {
...
+  do_not_include_profile: boolean;
+  supports_hub_url: boolean;
+  scopes: string[]
 };
 
 
 // new ecdsa private key each time
 const authTransitKey = crypto.randomBytes(32).toString('hex');
-const authTransitPubkey = getPublicKeyFromPrivateKey(authTransitKey);
+getPublicKeyFromPrivateKey(authTransitKey);
```
Comment (jcnelson) [inline src/auth.ts:105]: This line can just be deleted, since it looks like we don't even use this var anywhere.

**Review by jcnelson — CHANGES_REQUESTED** [review]
Thanks for cleaning this code up @zone117x!  I can't merge it yet, though, because I encountered a few show-stopping problems while testing this:

* `npm i` does _NOT_ install dependencies with the given package.json.  I had to delete the `preinstall` script, run `npm i`, and then add the `preinstall` script back.

* The resulting `lib/index.js` is not made executable as it should be.

* When testing the `authenticator` command, I get `"Unable to authenticate app request: Cannot read property 'mnemonicToSeed' of undefined`.  The `mnemonicToSeed` method is in `bip39`, and indeed when I run `node` from the command-line within the `cli-blockstack` directory with my `node_mdules` and then `require('bip39')`, I can access this method.  Somehow, it's not available during runtime.  Subsequent debugging revealed that while the module is being imported, the module itself is `undefined` at runtime.

I don't know what to make of the above.

EDIT:
```bash
$ node -v
v10.15.2
$ npm -v
6.10.2
$ env | grep NVM
NVM_DIR=/home/jude/.nvm
NVM_CD_FLAGS=
NVM_BIN=/home/jude/.nvm/versions/node/v10.15.2/bin
$ env | grep NODE
$ which npm
/home/jude/.nvm/versions/node/v10.15.2/bin/npm
$ which node
/home/jude/.nvm/versions/node/v10.15.2/bin/node
```

**Comment by zone117x** [comment]
@jcnelson this may have to due with bad package caching and/or linking. What do you have for `npm ls -g --depth=0` and `npm ls -g --depth=0 --link=true`?

**Comment by jcnelson** [comment]
```
$ npm ls -g --depth=0
/home/jude/.nvm/versions/node/v10.15.2/lib
├── blockstack-cli@1.0.0 -> /home/jude/Desktop/research/git/cli-blockstack
├── blockstack-subdomain-registrar@0.3.0 -> /home/jude/Desktop/research/git/blockstack-subdomain-registrar
├── generator-blockstack@0.5.0
├── npm@6.10.2
├── sqlite@3.0.3
├── typescript@3.5.3
└── yo@2.0.5

npm ERR! peer dep missing: eslint@^5.0.0, required by @typescript-eslint/eslint-plugin@1.13.0
npm ERR! peer dep missing: eslint@^5.0.0, required by @typescript-eslint/parser@1.13.0
```

```
$ npm ls -g --depth=0 --link=true
/home/jude/.nvm/versions/node/v10.15.2/lib
├── blockstack-cli@1.0.0 -> /home/jude/Desktop/research/git/cli-blockstack
└── blockstack-subdomain-registrar@0.3.0 -> /home/jude/Desktop/research/git/blockstack-subdomain-registrar

npm ERR! peer dep missing: eslint@^5.0.0, required by @typescript-eslint/eslint-plugin@1.13.0
npm ERR! peer dep missing: eslint@^5.0.0, required by @typescript-eslint/parser@1.13.0
```

**Comment by zone117x** [comment]
Hmm, I'm unable to reproduce the problem on any system, including after clearing out all module caches.
Just to clarify -- are you using `npm install -g https://github.com/blockstack/cli-blockstack#feature/package-updates` to install or another method?

**Comment by zone117x** [comment]
Pushed a minor change to simplify the scripts ran on `npm i -g [repo]` -- might get around any odd module caching/linking issues. LMK

**Comment by jcnelson** [comment]
This change fixed the first two issues, but the third one remains.

**Pull request merged**